core/src/memory/mod.rs

```diff
@@ -25,7 +25,7 @@ pub trait RWMem: ReadOnlyMem + WriteableMem {
     fn init(args: Self::InitArgs) -> Self;
 }
 
-pub type MemorySlotChange<Idx, MemVal> = (Idx, MemVal, MemVal);
+pub type MemorySlotChange<Idx, MemVal> = (Idx, MemVal);
 pub enum MemOpRecord<I, V> {
     Write(MemorySlotChange<I, V>),
 }
```

Comment:
Removed prev val from `MemorySlotChange`. Including prev val required being able to read from memory even if we were only writing to it just to specify the `MemOpRecord`. i.e. `MSTORE` only has to specify WritableMemory. Couldn't find a place we were using the prev val anyway.